pom.xml

```diff
@@ -687,7 +688,17 @@
   </reporting>
 
   <profiles>
-
+    <profile>
+        <!-- First, install the project without tests -> mvn install -DskipTests
+        This is a workaround for IntelliJ IDEA, see https://youtrack.jetbrains.com/issue/IDEA-148573
+        IDEA is able to recognize external artifacts with classifiers. But IDEA expects modules and their artifacts
+        without classifier. If the version differs from project, the idea would understand it as external artifact.
+        -->
+        <id>ide-development</id>
+        <properties>
+            <surefire-shared-utils.version>3-SNAPSHOT</surefire-shared-utils.version>
+        </properties>
+    </profile>
```

Comment:
I suppose that nobody will not install project with special profile ...

Reply:
@slawekjaranowski 
Slawomir, we were talking about it in previous PR. There is no other way. I was explaining it that the IDE and Maven work different way, the Maven understands shading and attaching artifacts but the IDEs don't and that's the reason why it has to be like this. We don't live in ideal world but we cannot wait forever until Jetbrain would fix the issue. If the developers does not care about red line, okay, but I don't because I do not want to have red lines in the code and also I do not want to copy paste the source code from external libraries because it is not maintainable way. So this is the explanation. Regarding the version 3-SNAPSHOT, it can be anything, that's only cosmetic issue.